src/olympia/devhub/tasks.py

```diff
@@ -143,17 +145,20 @@ def wrapper(task, id_, hash_, *args, **kw):
 
 
 @validation_task
-def validate_file_path(path, hash_, listed, **kw):
+def validate_file_path(path, hash_, listed=True, is_webextension=False, **kw):
     """Run the validator against a file at the given path, and return the
     results.
 
     Should only be called directly by ValidationAnnotator."""
+    run_linter = is_webextension and waffle.switch_is_active('addons-linter')
 
+    if is_webextension and run_linter:
```

Comment:
You don't need to repeat `is_webextension` in the `if`, since it's already a condition in `run_linter` above.